### DISCUSSION TITLE
Client Development Poll (客户端开发计划投票)

### DISCUSSION THREAD

**Discussion by SeanHeuc** [AlistGo/alist#4815] [Polls]
Dear community,

After some research, I think the most urgent requirement for alist is client app.  To use Alist truly as a cloud platform, people need to be able to access the data anywhere anytime with any devices. Not everyone always have a laptop/nas around, not everyone know how to set up a remote server, cloud flare, proxy, etc..

so here is the summary table I draw, let's take a poll.

## Client Development Plan

| Client Platform                       | Speed*                  | Dependency                                 | Compatibility                     | Comment                     | Available?                                                                                        |
|---------------------------------------|-------------------------|--------------------------------------------|-----------------------------------|-----------------------------|---------------------------------------------------------------------------------------------------|
| IOS/Mac                               | Max                     | Quantumult X/Surge/ShadowRocket [Paid App] | Browser, Any app work with webdav | data hijack for some url                 | work in progress*                                                                                 |
| Android                               | Max                     | Client App running in the background       | Browser, Any app work with webdav | client proxy data           |                                                                                                   |
| Mac/Windows/Linux/Docker              | Max                     | Application running in the background      | Browser, Any app work with webdav | client proxy data           |                                                                                                   |
| Web access                            | Limited by server       |                                            | Browser                           |                             | Yes                                                                                               |
| Web access[Alist Hybrid mode] | Max                     |                                            | Browser                           | use JS to make http request |                                                                                                   |
| Web access with alist-proxy           | Limited by proxy server | server with high bandwidth, domain         | Browser, Any app work with webdav | another server proxy data   | Yes with no Vault support                                                                         |
| Native IOS/Android App                       | Limited by server       |                                            | app internal function             | app has limited function    | [xlist](https://github.com/xlist-io/xlist) [AlistClient](https://github.com/BFWXKJGS/AlistClient) |

* 1 Assume client & cloud storage provider has unlimited bandwidth.
* 2 Details on the one I'm working on: after configured the rule in those apps, when accessing the alist server, the rule will get the 302 redirect url ,then apply the necessary header, and request the url from the device. 

--------------------
中文

经过一番研究，我认为对Alist的最紧迫需求是客户端应用。 要将Alist真正用作云平台，人们需要能够随时使用任何设备访问数据。 并非每个人随时周围有电脑或者nas，并不是每个人都知道如何设置远程服务器，cloud flare，流量代理等。

以下是我绘制的表格，请大家投票。

## 客户端开发计划

| 客户端平台                    | 速度*      | 依赖性                                    | 兼容性                | 其他               | 可用的？                                                                                                |
|--------------------------|----------|----------------------------------------|--------------------|------------------|-----------------------------------------------------------------------------------------------------|
| IOS/Mac                  | 最大       | Quantumult X/Surge/Shadowrocket [付费应用] | 浏览器，WebDav兼容的任何app | 特定url数据劫持        | 正在开发*                                                                                               |
| Android                  | 最大       | App后台运行                                | 浏览器，WebDav兼容的任何app | 客户端代理数据          |                                                                                                     |
| Mac/Windows/Linux/Docker | 最大       | 后台运行的应用程序                              | 浏览器，WebDav兼容的任何app | 客户端代理数据          |                                                                                                     |
| Web访问                    | 受服务器限制   |                                        | 浏览器                |                  | 是                                                                                                   |
| Web访问[Alis Hybrid模式]    | 最大       |                                        | 浏览器                | 使用浏览器脚本js来下载远程数据 |                                                                                                     |
| Web访问Alist-Proxy         | 受代理服务器限制 | 大带宽服务器，域名                            | 浏览器，WebDav兼容的任何app | 另一个服务器代理数据       | 是，但没有加密支持                                                                                           |
| 原生IOS/Android 应用               | 受服务器限制   |                                        | 应用程序内部功能           | 应用功能有限           | [xlist](https://github.com/xlist-io/xlist)   [AlistClient](https://github.com/BFWXKJGS/AlistClient) |

* 1 假设客户端和云存储提供商具有无限的带宽。
* 2 我正在开发的流量劫持规则：在这些vpn app中配置规则后，在访问ALIST服务器时，该规则将获得302重定向URL，然后修改数据包(header)后直接从设备请求URL。

大家投下票：

**Comment by Chenjun-Shen** [comment]
我没看懂，开发这个客户端的具体需求是什么，用这个客户端能解决用户什么痛点？

**Reply by SeanHeuc** [reply]
“人们需要能够随时使用任何设备访问数据”，但是很多设备在远程访问alist的使用体验不好，主要体现在速度慢。例如ios上如果远程连接alist webdav播放视频会很卡，不是每个人都有大宽带服务器架设cloud flare或者alist-proxy

**Reply by dissipator** [reply]
用户不关心用什么方法和技术，需要的是简单易用，怎么才不限速。最后我想有个功能是可选的加密备份，在云盘上看到时加密文件。

**Reply by Chenjun-Shen** [reply]
我也同意，好用就行。如果楼主有实力，可以先开发起来。

**Comment by pongfcnkl** [comment]
Android用termux挂一下就行了，不用开发APP
ios确实得开发一个

**Comment by earthjasonlin** [comment]
其实吧，感觉WebUI做的很好了

**Comment by yyqwe158** [comment]
个人认为，在ios平台的客户端，可以魔改alist的代码，从而增加挂载各类网盘的功能，如阿里云盘，蓝奏云等，以弥补在ios系统上无法本地搭建alist服务的缺点。
同时，目前的ios/安卓客户端可以增加部分新功能，如能进管理后台，记住常用的账号密码，记住2FA信息实现登陆时不用手输等。

**Reply by yyqwe158** [reply]
对于alist实际使用上，对于客户端的需求倒不是很大，首先，alist在浏览器中使用体验就很不错。此外，有一堆支持webdav的优秀应用，用支持webdav的文件管理器用alist，比其他做得跟🐶💩一样的网盘app舒服一亿倍。
